### DISCUSSION TITLE
Right Clicking Taskbar Icons

### DISCUSSION THREAD

**Discussion by dccjr3927** [valinet/ExplorerPatcher#3589] [Q&A]
Righting clicking task bar icon does not work. It has worked fine until  about a month ago. I have tried un-/re-installing. Here are my settings. Any idea what would be causing the issue now?

![image](https://github.com/user-attachments/assets/32d917fe-1218-472f-8b55-19112616764f)

**Comment by pyrates999** [comment]
ease post windows version and build.  You can find this by running winver.  Please include if you are running x86-64 or Arm.
Please post explorer patcher version.

Please include what settings you changed in explorer patcher properties application.  These should be screen shots.  If you can, please change it to english first.

Please include any other software you have installed that modifies the UI of windows.

Please show a screen shot of when you try to right click on the taskbar icons.

Try installing the latest pre-release: https://github.com/valinet/ExplorerPatcher/releases

**Comment by justM1ke001** [comment]
I'm having the same problem. W11 Pro, Version 23H2, build 22631.4037
ExplorerPatcher Version: 22621.3527.65.5
Taskbar Options:
- Taskbar style: Windows 10
- taskbar primary location: Top, taskbar secondary location: Bottom
- Screenshot of taskbar options: 
![image](https://github.com/user-attachments/assets/d2f986c7-d16f-47d6-8e62-830836791f46)


PROBLEM:
Right click program icon on taskbar produces a blank bar across that icon (image included).
If taskbar is located at Bottom or Left side, right-click works as expected, produces alternate menu.
Right click functions correctly with taskbar location: left. I did not test with location: right

- Update does not list any updates available.
- Adding ep_taskbar.2.dll made no difference.

Here is the screenshot image of the problem:
<img width="516" alt="explorer patcher bug 082424" src="https://github.com/user-attachments/assets/f2436a9b-4b4b-4b71-af41-2f3af3444e7f">

**Reply by pyrates999** [reply]
Read my comment and go to the url to download the latest pre-release of EP.

**Comment by bobster151** [comment]
I have this issue as well, 22621.3880.66.4 installed.

The problem only occurs when the Taskbar is on top, right click works when it is at the bottom or left/right.

**Reply by pyrates999** [reply]
Read my comment and go to the url to download the latest pre-release of EP.

**Reply by bobster151** [reply]
What is the pre-release version number?

**Reply by pyrates999** [reply]
look for the one with the label pre-release.  Should be 66.4.

**Comment by johnkarlhill** [comment]
Mine started doing this again as well.. but I think I know why.
The pop up menu always shows above the taskbar.
When the taskbar is on the side or bottom of the screen, no issues because we can see it.
On the top, however, it looks like nothing is happening (or Windows says "you can't have a menu there").

![image](https://github.com/user-attachments/assets/5875fbb5-a46c-41d7-bff3-a8dd0e6a59c0)

**Reply by bobster151** [reply]
But it used to work fine until recently.  When it first happened, I uninstalled ExplorerPatcher and re-installed it which fixed the issue for a while but when it happened again yesterday, this solution did not work.

Windows 11 Version 23H2 (OS Build 22631.4112)

**Reply by pyrates999** [reply]
Read my comment and go to the url to download the latest pre-release of EP.

**Comment by justM1ke001** [comment]
I did read the response from pyrTES999, see that the problem is because of a change in Windows. I added the Defender antivirus exclusions using the powershell script as directed. The exclusions were accepted and appear in the Defender exclusion list.
However, any attempt at downloading the 22621.3880.66.4 pre-release version of ep_setup from the assets list results in a series of error messages:
![image](https://github.com/user-attachments/assets/06880d5f-a72c-4ff9-b6d6-cbb6750af9b1)

**Reply by justM1ke001** [reply]
I also received this error message, although it did not appear in subsequent download attempts.
![image](https://github.com/user-attachments/assets/c92e3f76-95bd-4a92-b2b2-5cdaa2c79adf)

**Reply by pyrates999** [reply]
You need to exclude the directory in windows defender that you download ep_setup.exe to as well.

**Comment by silvervirus** [comment]
![image](https://github.com/user-attachments/assets/bbc322c7-bfdc-46b2-a230-ee7c3e4f3a31)
i have the latest preview version still no right click. pretty sure its opening above the bar because left ,right and down all work so top might just be opening above where we can't see it.
[taskbar.webm](https://github.com/user-attachments/assets/5d492338-f4ea-4c58-ba45-e55d85e47603)  a workaround is use Shift plus Right click can help open the program if that what your looking for with your rightclick

**Comment by Doigan** [comment]
I'm also using the taskbar at the top and 22621.3880.66.5 solved this issue for me. It now works as before.

**Reply by bobster151** [reply]
I'm using that version and it is not working, would you mind exporting your settings so I can try them?

**Reply by pyrates999** [reply]
Please include what settings you changed in explorer patcher properties application.  These should be screen shots.  If you can, please change it to english first.

**Reply by bobster151** [reply]
I can't remember all the changes I made in Properties, there are quite a few options!

This is what I have for the TaskBar section. 
![image](https://github.com/user-attachments/assets/f16bd4ab-2c99-452a-b2ff-f8cd428ba889)

**Reply by justM1ke001** [reply]
22621.3880.66.5 fixed the issue for me, too.

**Reply by pyrates999** [reply]
After you update to latest pre-release of EP, if you want to use the windows 10 taskbar, do the following:

Open up EP Properties application, go to taskbar, and set **taskbar style** to **Windows 10 (ExplorerPatcher)**, _not_ **Windows 10**.  We no longer offer more support on the stock Windows 10 taskbar.

**Reply by bobster151** [reply]
Thank you, I have changed that property and it is working as expected now.

**Comment by justM1ke001** [comment]
After updating to 22621.3880.66.5, which came via the Check for Update option, not from the repository, the Windows 10 taskbar/Top position options worked as expected.

Is the Windows10 (ExplorerPatcher) option more robust in some way? I assume it is, but I don't normally change things when they work.

**Reply by pyrates999** [reply]
please see this: https://github.com/valinet/ExplorerPatcher/wiki/ExplorerPatcher's-taskbar-implementation

**Comment by flush2x** [comment]


**Reply by pyrates999** [reply]
windows updates usually require you to reboot, so reboot after installing it.

**Reply by flush2x** [reply]
Is that your fix? Did you read the post in it's entirety? 

That would put the machine at Window 11, OS Build 22631.4391, which would no longer allow me to right click on the start menu.

**Reply by pyrates999** [reply]
you said 

> I can right click on the start menu on another machine that hasn't rebooted after installing 2024-10 Cumulative Update Preview for Windows 11 Version 23H2 for x64-based Systems (KB5044380). I don't plan on rebooting.

and installing the above requires a reboot, so reboot.

**Reply by flush2x** [reply]
Rebooting will then remove the ability to right click on the start menu. 

That is why I'm not rebooting.

Do you want me to not be able to right click on the start menu?

**Reply by pyrates999** [reply]
see here why: https://www.reddit.com/r/explainlikeimfive/comments/s40a9a/comment/hso12xt/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button

**Reply by pyrates999** [reply]
if you don't want preview updates to be installed:

In the settings app, untick "Get the latest updates as soon as available" so that preview updates are not automatically installed.

All preview updates are then shown as download and install and are never automatically installed if you have the above unticked.

Preview updates are also not fully tested by Microsoft and thus could have bugs.

You should only be installing preview updates if you are a developer or are OK with running pre-release versions of EP.
